### ISSUE TITLE
Cryptic error message when instantiating type that fails to implement interface methods

### ISSUE THREAD

**Issue by jvandort** [gradle/gradle#27108]
### Current Behavior

Consider the following minimal reproducer:
```
given:
buildFile << """
    abstract class Reproducer implements Named {
        @Inject
        Reproducer() {}
    }
    objects.newInstance(Reproducer)
"""

expect:
succeeds("help")
```

This fails with:
```
	at org.gradle.internal.instantiation.generator.DependencyInjectingInstantiator.doCreate(DependencyInjectingInstantiator.java:69)
	at org.gradle.internal.instantiation.generator.DependencyInjectingInstantiator.newInstance(DependencyInjectingInstantiator.java:55)
	at org.gradle.api.internal.model.DefaultObjectFactory.newInstance(DefaultObjectFactory.java:90)
	at build_1slav2hx8m13lbvv1i8qw5ki6.run(gradle/platforms/software/dependency-management/build/tmp/teŝt files/Reproducer/fs6a6/build.gradle:6)
	at org.gradle.groovy.scripts.internal.DefaultScriptRunnerFactory$ScriptRunnerImpl.run(DefaultScriptRunnerFactory.java:91)
	... 150 more
Caused by: java.lang.reflect.GenericSignatureFormatError: Signature Parse error: expected '('
	Remaining input: Ljava/lang/String;()V
	at org.gradle.internal.instantiation.generator.AbstractClassGenerator$GeneratedClassImpl$GeneratedConstructorImpl.getGenericParameterTypes(AbstractClassGenerator.java:554)
	at org.gradle.internal.instantiation.generator.DependencyInjectingInstantiator.addServicesToParameters(DependencyInjectingInstantiator.java:162)
	at org.gradle.internal.instantiation.generator.DependencyInjectingInstantiator.convertParameters(DependencyInjectingInstantiator.java:121)
	at org.gradle.internal.instantiation.generator.DependencyInjectingInstantiator.doCreate(DependencyInjectingInstantiator.java:62)
	... 154 more
```

### Expected Behavior

An error message that better guides the user to a solution

### Steps to Reproduce

See above

### Gradle version

8.4

**Comment by cobexer** [comment]
Thank you for providing a valid report.

The issue is in the backlog of the relevant team, but this area of Gradle is currently not a focus one, so it might take a while before a fix is made.

**Comment by nbrugger-tgm** [comment]
Is there a workaround for this issue? What does this error mean?
I tried re-writing the class in Java, but that didn't help the error is the same.

Is it known to the Gradle team what the correct message would be, and what would be needed to be done to correct this?
I know this will be fixed in the next RC 8.7, but it would be nice to have a quick workaround here

**Comment by bamboo** [comment]
> Is there a workaround for this issue? What does this error mean?

This is a bug in Gradle (the bogus error message) with no workaround but it's triggered by the missing `String` argument in the `newInstance` call representing the name of the `Named` instance, i.e., the following should work:

```
// pass the name of the object explicitly
objects.newInstance(Reproducer, 'foo')
```
